azure/services/aso/tags.go

```diff
@@ -42,14 +47,29 @@ func reconcileTags(t TagsGetterSetter, existing genruntime.MetaObject, parameter
 				return errors.Wrapf(err, "failed to unmarshal JSON from %s annotation", tagsLastAppliedAnnotation)
 			}
 		}
-	}
 
-	existingTags, err := t.GetActualTags(existing)
-	if err != nil {
-		return errors.Wrapf(err, "failed to get actual tags for %s %s/%s", existing.GetObjectKind().GroupVersionKind(), existing.GetNamespace(), existing.GetName())
+		var err error
```

Comment:
I see you had to add this code, but it ends up differing a fair amount from the source PR. Was there another PR we should have cherry-picked or something? Just curious; the changes look sane.

Reply:
Yeah I think the ASO generics PR conflicted with this a bit: #3997